### PR TITLE
Cleanup the upgrade code

### PR DESCRIPTION
**Pull request by jsuchome**
Cleanup of some minor stuff, DRY, etc.

Touches these cards 
https://trello.com/c/a630zeyt/347-move-save-substep-inside-the-methods-executing-the-substeps
https://trello.com/c/fGtuEsRe/343-no-comput-enodes-controller-substep-remains-running-atfter-the-whole-upgrade

Not really tested - I think the part with moving those substeps needs some testing to see if I broke something...
UPDATE: Tested manually

### CODE REVIEW

File: crowbar_framework/app/models/api/upgrade.rb [hunk 1]
```diff
@@ -699,59 +713,32 @@ def nodes(component = "all")
...
-            ::Crowbar::UpgradeStatus.new.end_step
-          end
+        # if all nodes are upgraded, cleanup and end the whole step
+        status = ::Crowbar::UpgradeStatus.new
+        if status.progress[:remaining_nodes].zero?
+          status.save_substep(substep, :finished)
+          finalize_nodes_upgrade
+          status.end_step
         end
+        true
```
Comment (Itxaka) [inline crowbar_framework/app/models/api/upgrade.rb]: Sorry, I cant see where is this used or what is used for :(

Comment (jsuchome) [inline crowbar_framework/app/models/api/upgrade.rb]: Good point, it wasn't there before. I tried to add it to appease my failing testsuite - I still don't see why it is suddenly needed.

I'll need to investigate then...

Comment (jsuchome) [inline crowbar_framework/app/models/api/upgrade.rb]: I was hoping nobody will spot it ;-)

Comment (jsuchome) [inline crowbar_framework/app/models/api/upgrade.rb]: Found it. Testsuite update will follow

Comment (Itxaka) [inline crowbar_framework/app/models/api/upgrade.rb]: > I was hoping nobody will spot it ;-)
:rofl:

**Pull request merged**